python/cudf/cudf/tests/test_groupby.py

```diff
@@ -381,19 +381,76 @@ def emulate(df):
 
 
 @pytest.fixture(scope="module")
-def groupby_jit_data():
+def groupby_jit_data_small():
+    """
+    Small dataset for testing groupby apply with jit
+    returns a dataframe whose keys columns define
+    4 groups of size 1, 2, 3, 4 as well as an additional
+    key column that can be used to test subgroups
+
+    useful for very basic testing of result values
+
+    """
     np.random.seed(0)
```

Comment:
I realize it's not quite in this PR, but having the seed be set as part of a fixture seems like a confusing and potentially misleading side effect. If you want the fixture to generate a consistent output, can you use a seeded generator instead? Here are the docs on the canonical way to do this: https://numpy.org/doc/stable/reference/random/generator.html